hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java

```diff
@@ -1935,8 +1935,8 @@ protected Subject getSubject() {
   @InterfaceStability.Evolving
   public <T> T doAs(PrivilegedAction<T> action) {
     if (LOG.isDebugEnabled()) {
-      LOG.debug("PrivilegedAction [as: {}][action: {}]", this, action,
-          new Exception());
+      LOG.debug("PrivilegedAction [as: {}][action: {}][from: {}]", this, action,
+              Thread.currentThread().getStackTrace()[2]);
```

Comment:
we want the whole stack. If you are worried about interpretation, just create an exception with a string like 
```
new Exception("Stack trace for debugging")
```
same below